packages/types/src/getters/value-view.ts

```diff
@@ -9,3 +10,26 @@ export const getMetadata = createGetter((valueView?: ValueView) =>
 export const getDisplayDenomExponentFromValueView = getMetadata.pipe(getDisplayDenomExponent);
 
 export const getAssetIdFromValueView = getMetadata.pipe(getAssetId);
+
+export const getAmountFromValueView = createGetter((valueView?: ValueView) =>
+  valueView?.valueView.value ? valueView.valueView.value.amount : undefined,
+);
+
+export const getDisplayDenomFromView = createGetter((view?: ValueView) => {
+  if (view?.valueView.case === 'unknownAssetId') {
+    if (!view.valueView.value.assetId) throw new Error('no asset id for unknown denom');
```

Comment:
Hmm... I think we should return `undefined` here, so that this getter still works with `.optional()`.